cranelift/assembler/meta/src/generate/inst.rs

```diff
@@ -129,15 +171,19 @@ impl dsl::Inst {
 
     /// `impl Debug for <inst> { ... }`
     pub fn generate_display_impl(&self, f: &mut Formatter) {
-        let struct_name = self.struct_name();
+        let immediate_name = self.immediate_name();
+        let struct_name = self.struct_name() + &immediate_name;
         fmtln!(f, "impl std::fmt::Display for {struct_name} {{");
         f.indent_push();
         fmtln!(f, "fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {{");
 
         f.indent_push();
+        let mut i = 0;
         for op in self.format.locations() {
-            let to_string = op.generate_to_string();
+            let extension = self.format.operands[i].extension;
+            let to_string = op.generate_to_string(extension);
```

Comment:
I think we talked about this: if we iterate on `self.format.operands.iter()` here and move the `generate...` method up a level to `Operand` then we don't need to do the extra `i += 1` dance.